
https://bugzilla.redhat.com/show_bug.cgi?id=798636

--- neon-0.29.3/src/ne_session.c.sni
+++ neon-0.29.3/src/ne_session.c
@@ -183,8 +183,10 @@ ne_session *ne_session_create(const char
             ia = ne_iaddr_parse(hostname, ne_iaddr_ipv6);
 
         if (ia) {
-            sess->flags[NE_SESSFLAG_TLS_SNI] = 1;
             ne_iaddr_free(ia);
+        } 
+        else {
+            sess->flags[NE_SESSFLAG_TLS_SNI] = 1;
         }
         NE_DEBUG(NE_DBG_SSL, "ssl: SNI %s by default.\n",
                  sess->flags[NE_SESSFLAG_TLS_SNI] ?
